indico/modules/events/management/controllers/settings.py

```diff
@@ -65,6 +66,11 @@ def _process(self):
                            .join(Room)
                            .has_rows())
             show_booking_warning = not has_overlap
+        if not ReferenceType.query.has_rows():
+            has_reference_type = False
+        else:
+            has_reference_type = True
+        self.event.has_reference_type = has_reference_type
```

Comment:
never set random stuff on the event object! pass it as a kwarg to the template instead. (also, it should be plural `has_reference_types`)